ui/pages/confirmations/components/simulation-details/formatAmount.ts

```diff
@@ -80,6 +80,18 @@ export function formatAmount(locale: string, amount: BigNumber): string {
     MAX_SIGNIFICANT_DECIMAL_PLACES - digitsLeftOfDecimal + 1,
   );
 
+  if (maximumFractionDigits === 0) {
+    return new Intl.NumberFormat(locale, {
+      maximumFractionDigits,
+    } as Intl.NumberFormatOptions).format(
+      // string is valid parameter for format function
+      // for some reason it gives TS issue
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/format#number
+      amount.toFixed(0) as unknown as number,
```

Comment:
Awesome find!

In that case, we don't need the `if (maximumFractionDigits === 0) {` above... we can have the same code path for all amounts here.